NOTES.WIN

```diff
@@ -2,8 +2,35 @@
  NOTES FOR THE WINDOWS PLATFORMS
  ===============================
 
- Requirement details for native (Visual C++) builds
- --------------------------------------------------
+ Windows targets can be classified as "native", ones that use Windows API
+ directly, and "hosted" which rely on POSIX-compatible layer. "Native"
+ targets are VC-* [where "VC" stems from abbreviating Microsoft Visual C
+ compiler] and mingw[64]. "Hosted" platforms are Cygwin and MSYS[2]. Even
+ though the latter is not directly supported by OpenSSL Team, it plays
+ essential role in supporting MinGW targets. In the nutshell MinGW builds
+ are always cross-compiled. On Linux and Cygwin they look exactly as such
+ and require --cross-compile-prefix option. While on MSYS[2] it's solved
+ rather by placing gcc that produces "MinGW binary" code 1st on $PATH.
+ This is customarily source of confusion. "Hosted" applications "live" in
+ emulated file system name space with POSIX-y root, mount points, /dev
+ and even /proc. Confusion is intensified by the fact that MSYS2 shell
+ [or rather emulated execve(2) call] examines the binary it's about to
+ start, and if it's found *not* to be linked with MSYS2 POSIX-y thing,
+ it translates command line arguments that look like file names from
+ emulated name space to "native". For example '/c/some/where' becomes
+ 'c:\some\where', '/dev/null' - 'nul'. This creates an illusion that
+ there is no difference between MSYS2 shell and "MinGW binary", but
+ there is. Just keep in mind that "MinGW binary" "experiences" Windows
+ system in exactly same way as one produced by VC, and in its essence
+ is indistinguishable from the latter. [Which by the way is why
+ it's referred to in quotes here, as "MinGW binary", it's just as
+ "native" as it can get.]
```

Comment:
Oh!  There's one thing I've notice in your style of writing that comes out as ...  odd.  You use squares parentheses [such as this] when one would usually use round parentheses (such as these).

It becomes a bit extra confusing when you have things like `mingw[64]` and `MSYS[2]`, where the part in square brackets are optional.

Would you mind terribly following the norm re the use of parentheses in written natural languages (or more specifically, English)?  Note that I'm only asking this for things that *are* natural language (so no one thinks I'm trying to say that `mingw[64]` should be changed to `mingw(64)`...  I'm certainly not saying that!)

Reply:
I suppose I mechanically extend that "manual page" convention, when [] denote something optional, or something that can be omitted, to sentences. But sure, I'll fix. (As more feedback provided. And if not, then just fix after some grace period.) Like this, right?

Reply:
(yes, exactly like that)